Makefile

```diff
@@ -267,7 +267,12 @@ rpm:
 
 	# Copy spec file to rpmbuild specs directory
 	mkdir -p $(RPMBUILDDIR)/SPECS
-	sed -e 's/@version@/$(PACKAGE_VERSION)/g; s/@release@/$(PACKAGE_RELEASE)/g' contrib/centos/aziot-identity-service.spec >$(RPMBUILDDIR)/SPECS/aziot-identity-service.spec
+
+	# Ask openssl for its engine dir and on versions <1.1 fallback to a hardcoded path.
+	which openssl # Assert that openssl exists and that the empty output of `openssl version -e` is not because it never even ran
+	ENGINESDIR="$$(openssl version -e 2> /dev/null | sed 's/^ENGINESDIR: "\(.*\)"$$/\1/')"; \
+	ENGINESDIR="$${ENGINESDIR:-%\{_libdir\}/openssl/engines}"; \
+	sed -e "s/@version@/$(PACKAGE_VERSION)/g; s/@release@/$(PACKAGE_RELEASE)/g; s|@enginesdir@|$$ENGINESDIR|g" contrib/centos/aziot-identity-service.spec >$(RPMBUILDDIR)/SPECS/aziot-identity-service.spec
```

Comment:
This is insufficient. Apart from using a different enginesdir, openssl 1.1 also requires the engine .so to not have a leading `lib` prefix. I'm fixing it on master as part of fixing E2E tests, so wait for that.